### PR TITLE
[GOBBLIN-631] Add option to use timezone for TimeAwareDatasetFinder

### PR DESCRIPTION
**Pull request by amarnathkarthik**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN-631) issues and references them in the PR title. For example, "[GOBBLIN-631] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-631

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
        Added option to use timezone for TimeAwareRecursiveCopyableDataset

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Existing Test case passed

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by amarnathkarthik** [comment]
@sv2000 Added an option to configure timezone (org.joda.time.DateTimeZone) to LocalDateTime for TimeAwareRecursiveCopyableDataset. Defaulted to PT timezone, if timezone property not defined.

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java [hunk 1]
```diff
@@ -36,15 +37,20 @@
 
 import com.google.common.collect.Lists;
 
+import org.apache.gobblin.configuration.ConfigurationKeys;
+
 public class TimeAwareRecursiveCopyableDataset extends RecursiveCopyableDataset {
   private static final String CONFIG_PREFIX = CopyConfiguration.COPY_PREFIX + ".recursive";
   public static final String DATE_PATTERN_KEY = CONFIG_PREFIX + ".date.pattern";
   public static final String LOOKBACK_TIME_KEY = CONFIG_PREFIX + ".lookback.time";
+  public static final String DEFAULT_DATE_TIME_PATTERN_TIMEZONE = ConfigurationKeys.PST_TIMEZONE_NAME;
```
Comment (sv2000) [inline gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java]: Can you call it DEFAULT_DATE_PATTERN_TIMEZONE?

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java [hunk 2]
```diff
@@ -36,15 +37,20 @@
...
+  public static final String DATE_TIME_PATTERN_TIMEZONE_KEY = CONFIG_PREFIX + ".datetime.timezone";
```
Comment (sv2000) [inline gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java]: Same as above: DATE_PATTERN_TIMEZONE_KEY?

**Comment by sv2000** [comment]
@amarnathkarthik The travis build failed. Can you take a look?

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java (hunk 2 above)
Comment (amarnathkarthik) [reply gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java]: Changes variable name

**Comment by amarnathkarthik** [comment]
@sv2000 Changes variable names

**Comment by amarnathkarthik** [comment]
@sv2000 Test passed, Please review. Thanks

**Comment by htran1** [comment]
+1

**Pull request closed without merging**